### PR TITLE
[pytorch][ec2][sagemaker][build][test] patched binaries and new armcl install

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
* Patching binaries 
* using a different install method for ARM-CL to allow images to build faster
* Updated DLAMI setup for DLC tests from https://github.com/aws/deep-learning-containers/pull/2437 to support graviton tests
* Pinning "package" in SageMaker test pip requirements to 20.1 as 22.0 dropped LegacyVersion https://github.com/pypa/packaging/pull/407, which is used for inference test commands. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [x] Revision B: `sagemaker_remote_tests = "rc"`
- [x] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [X] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/conftest.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ def delete_s3_artifact_copy():
     request.addfinalizer(delete_s3_artifact_copy)
 
     ec2_utils.install_python_in_instance(conn, python_version="3.9")
-
+
```
[the remark this answers is not in the document]
Comment (xncqr) [reply test/dlc_tests/conftest.py]: I saw that and it's removed

File: test/test_utils/ec2.py [hunk 2]
```diff
@@ -958,8 +958,15 @@ def install_python_in_instance(context, python_version="3.9"):
...
-        # Need to configure PATH and PYENV_ROOT changes in alternative location because ~/.bashrc and ~/.profile are
-        # not used when using bash through Fabric Connection.
+
+        # for images that do not have /etc/profile.d/dlami.sh, we will make it here
+        if context.run("test -f /etc/profile.d/dlami.sh", warn=True, hide=True).failed:
+            LOGGER.info("/etc/profile.d/dlami.sh does not exist. Making...")
+            context.run("sudo touch /etc/profile.d/dlami.sh")
+            LOGGER.info("adding /etc/profile.d/dlami.sh to .bashrc")
+            context.run("""echo '[ -z "$PS1" ] && source /etc/profile.d/dlami.sh'|cat - .bashrc > ./temprc """
+                        """&& mv ./temprc ./.bashrc""", hide=True)
```
[the remark this answers is not in the document]
Comment (xncqr) [reply test/test_utils/ec2.py]: Updated as match. initial build was to see if the setup worked, now cleaning it.

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -61,8 +61,8 @@ def get_ami_id_ssm(region_name, parameter_path):
 AML2_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning Base AMI (Amazon Linux 2) Version ??.?")
 AML2_BASE_DLAMI_US_EAST_1 = get_ami_id_boto3(region_name="us-east-1", ami_name_pattern="Deep Learning Base AMI (Amazon Linux 2) Version ??.?")
 # We use the following DLAMI for MXNet and TensorFlow tests as well, but this is ok since we use custom DLC Graviton containers on top. We just need an ARM base DLAMI.
-UL20_CPU_ARM64_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI Graviton GPU PyTorch 1.10.0 (Ubuntu 20.04) ????????")
-UL20_CPU_ARM64_US_EAST_1 = get_ami_id_boto3(region_name="us-east-1", ami_name_pattern="Deep Learning AMI Graviton GPU PyTorch 1.10.0 (Ubuntu 20.04) ????????")
+UL20_CPU_ARM64_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI Graviton GPU CUDA 11.4.2 (Ubuntu 20.04) ????????")
```
[the remark this answers is not in the document]
Comment (xncqr) [reply test/test_utils/__init__.py:64]: instead of using framework AMI, I'm using the most generic DLAMI we have as it's used for testing for both PT and TF (smaller and no framework on the AMI). The same process we do with all the x86_64 tests (example "Deep Learning Base AMI (Amazon Linux 2) Version ??.?"). The test setup update would allow us to use even a basic Ubuntu AMI as this time, but sticking with a DLAMI.

**Review by kenny-ezirim — DISMISSED** [review]

File: test/test_utils/ec2.py [hunk 4]
```diff
@@ -958,8 +958,15 @@ def install_python_in_instance(context, python_version="3.9"):
     """
     if context.run("pyenv --version", warn=True, hide=True).failed:
         context.run("""ls ~/.pyenv || git clone https://github.com/pyenv/pyenv.git ~/.pyenv""", hide=True)
-        # Need to configure PATH and PYENV_ROOT changes in alternative location because ~/.bashrc and ~/.profile are
-        # not used when using bash through Fabric Connection.
+
+        # for images that do not have /etc/profile.d/dlami.sh, we will make it here
```
Comment (kace) [inline test/test_utils/ec2.py:962]: 👏🏾 Nice!

**Review by saimidu — APPROVED** [review]
Looks good! Please run the standard sagemaker tests as well, and then feel free to merge.

**Comment by saimidu** [comment]
There are other failures in the scan tests. Please take a look at the failing sanity test job.

**Review by kenny-ezirim — APPROVED** [review]
